channel/adjudicator_sub.go

```diff
@@ -0,0 +1,188 @@
+// Copyright 2023 - See NOTICE file for copyright holders.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package channel
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gorilla/websocket"
+	"net/url"
+	gpchannel "perun.network/go-perun/channel"
+	"perun.network/perun-cardano-backend/channel/types"
+	"perun.network/perun-cardano-backend/wire"
+)
+
+// AdjudicatorSub is a subscription to the Adjudicator events.
+// Instances should only be created using PAB.NewSubscription.
+type AdjudicatorSub struct {
```

Comment:
We can decide **NOT** to export the specific type. This would disallow a user of this lib to even create this struct without going through `PAB.NewSubscription`.

```suggestion
type adjudicatorSub struct {
```

Reply:
I thought about this for quite a while.
I would like to avoid this, because I want the PAB's functions `NewInternalSubscription` and `NewPerunEventSubscription` to specifically have as return type our `AdjudicatorSub`type instead of `go-perun`'s `channel.AdjudicatorSubscription` interface. 
I want this to be the case so that our backend's documentation and implementation is linked if someone writes code that uses this functionality. For example, `go-perun`'s `AdjudicatorSubscription` interface does not specify what `Next` does if it is called **after** a call to `Next` returned `nil` once.
Also a Fix to another comment requires a user to only ever interact with an `AdjudicatorSub` from within a single go-routine.